subprojects/gradle/namespaced-resources-fixer/src/main/kotlin/com/avito/android/plugin/FileResourcesFixer.kt

```diff
@@ -0,0 +1,169 @@
+
+        val mergedRClassImport = findMergedRImport(ktFile, app) ?: return null
+
+        val mergedRClassReferencesFinder = MergedRClassReferencesFinder(mergedRClassImport)
+        mergedRClassReferencesFinder.visitKtFile(ktFile)
+
+        val references = mergedRClassReferencesFinder.references
+        if (references.isEmpty()) return null
+
+        val uniqueResources: List<ResourceId> = findUniqueResources(libraries) // TODO: support string, drawable, ...
+
+        val replacedModules = replaceReferences(references, uniqueResources)
+        fixImports(ktFile, replacedModules)
+
+        return ktFile.text
+    }
+
+    private fun replaceReferences(
+        rClassReferences: List<PsiElement>,
+        uniqueResources: List<ResourceId>
+    ): Set<AndroidModule> {
+        val replacedModules = mutableSetOf<AndroidModule>()
+        rClassReferences.forEach { refElement ->
+            val resource = uniqueResources.find { resource ->
+                refElement.text.endsWith(".id.${resource.id}")
+            }
+            if (resource != null) {
+                val newReference = createElementForResourceRef(resource)
+                refElement.replace(newReference)
+                replacedModules.add(resource.module)
+            }
+        }
+        return replacedModules
+    }
+
+    private class MergedRClassReferencesFinder(
+        private val classImport: KtImportDirective
+    ) : KtTreeVisitorVoid() {
+
+        var references = mutableListOf<PsiElement>()
+
+        override fun visitElement(element: PsiElement) {
+            if (element.isReferenceTo(classImport)) {
+                references.add(element)
+            }
+            super.visitElement(element)
+        }
+
+        private fun PsiElement.isReferenceTo(importElement: KtImportDirective): Boolean {
+            val element = this
+            val isDotReferenceExpression = element is KtDotQualifiedExpression
+            val partialDotReferenceExpression = element.context is KtDotQualifiedExpression
+            val insideImport = element.context is KtImportDirective
+
+            return isDotReferenceExpression
+                && !insideImport
+                && !partialDotReferenceExpression
+                && element.text.startsWith(importElement.referencedClass + ".")
+        }
+
+        private val KtImportDirective.referencedClass: String
+            get() = if (aliasName == null) {
+                children.last().children[1].text
+            } else {
+                aliasName!!
+            }
+    }
+
+    private fun findMergedRImport(file: KtFile, app: AndroidModule): KtImportDirective? {
+        val imports: List<KtImportDirective> = file.importList?.imports?.toList() ?: return null
+        return imports.firstOrNull {
+            it.importPath.toString().startsWith(app.packageId)
+        }
+    }
+
+    private fun findUniqueResources(modules: List<AndroidModule>): List<ResourceId> {
```

Comment:
Is it class from detekt?

Reply:
No, they are just DTOs in this plugin:

```kotlin
internal data class AndroidModule(val packageId: String, val ids: List<String>)

internal data class ResourceId(val id: String, val module: AndroidModule)
```

Reply:
In short, these models describe Gradle projects.